FIX: Fixed ISXB-586 for 2022.3.69f1 and up.

benoitalain opened the pull request:
### Description

Original PR: https://github.com/Unity-Technologies/InputSystem/pull/1975

Jira ticket: https://jira.unity3d.com/browse/ISXB-586
2022.3 PR: https://github.cds.internal.unity3d.com/unity/unity/pull/81547

2022.3 PR has successfully landed before this PR.

### Testing status & QA

Adds 2022.3.69f1 to `#define` flag `UNITY_INPUT_SYSTEM_SENDPOINTERHOVERTOPARENT` which was used in original fix.

From the original fix:
Implements BaseInputModule sendPointerHoverToParent property for InputSystemUIInputModule. Also properly set PointerEventData reentered and fullyExited properties.

### Overall Product Risks

- Complexity: 0
- Halo Effect: 1

### Comments to reviewers

I did not do any kind of testing whatsoever on this backport fix.

### Checklist

Before review:

- [ ] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - JIRA ticket linked, example ([case %<ID>%](https://issuetracker.unity3d.com/product/unity/issues/guid/<ID>)). If it is a private issue, just add the case ID without a link.
    - Jira port for the next release set as "Resolved".
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

After merge:

- [ ] Create forward/backward port if needed. If you are blocked from creating a forward port now please add a task to ISX-1444.

benoitalain committed:
Fixed ISXB-586 for 2022.3.69f1 and up.

# Conflicts:
#	Packages/com.unity.inputsystem/CHANGELOG.md

benoitalain commented:
@Pauliusd01 I added you for QA. Could you verify
* 2022.3.68f1 and below don't have the fix but also don't get compile errors when using InputSystem package with this branch
* 2022.3.69f1 has the fix
* 6000.0.14f1 and below don't have the fix but also don't get compile errors when using InputSystem package with this branch
* 6000.0.15f1 or more still has the fix

Or test 2022.3.69f1 and for other versions just trust the process that we'll get bug reports if I regressed anything. The risk of that is pretty low.

benoitalain committed:
Fixed wrong #define expression.

ericksson commented:
Cancelling all jobs running on this branch due to Android jobs being broken and stuck - see https://unity.slack.com/archives/C94RMJJ5T/p1760644095487229

Follow that thread to see when Lukas has a fix for the issue. Please refrain from rerunning the tests until the fix is available and you have it merged in this branch.

jfreire-unity committed:
Merge branch 'develop' into 2022.3/isxb-586-inputsystemsendpointerhovertoparent

Pauliusd01 commented:
Taking a look today

Pauliusd01 reviewed (changes requested):
Here's what I've checked so far and I'm getting errors on 6.0.14f1:

2022.3.68f1 - no fix for both, no errors or other issues for both in playmode or player
2022.3.69f1 - no fix 1.15.0, fixed on PR 1.15.1, no errors or other issues for both in playmode or player
6000.0.14f1 - no fix 1.15.0, **errors on PR 1.15.1**

<img width="666" height="273" alt="image" src="https://github.com/user-attachments/assets/eb106e86-3c06-47f0-9f2f-3f9213402f56" />

Editor used:
```
Unity 6 Preview
6000.0.14f1.6822.6000.0/release.4172
Revision: 6000.0/release e12ab2d6a089
```

jfreire-unity committed:
Limit define for 2022.3

Extended it until 2023.0.0 which doesn't seem to exist anymore.

This makes sure that from 6.0.0 until 6.0.14 the define doesn't exist.

jfreire-unity commented:
@Pauliusd01 please try again. I believe my last commit fixes it

jfreire-unity committed:
Merge branch 'develop' into 2022.3/isxb-586-inputsystemsendpointerhovertoparent

jfreire-unity commented:
For awareness, this closes https://jira.unity3d.com/browse/ISXB-1296
